### PR TITLE
Add support for post-CI automated benchmarking

### PR DESCRIPTION
**Pull request by FUDCo**

### CODE REVIEW

**Review by warner — APPROVED** [review]
LGTM

**Comment by FUDCo** [comment]
The debugging gymnastics I had to go through to get this to pass CI testing were helpful because it got the implementation of the code to the point where it executes the benchmark correctly all the way through.  However, the fact that I _had_ to do that work to get it to pass means that it's running the benchmarks on every push instead of only on merges with `master`, so we've still got work to do before this can land.

**Comment by warner** [comment]
@michaelfig could you take a quick look at the cache-restore part of the GitHub Actions script? I don't trust my knowledge how that works, would appreciate a second pair of eyeballs

**Review by warner — APPROVED** [review]

**Review by michaelfig — CHANGES_REQUESTED** [review]
I can't give a full review right now due to time constraints.

I respectfully request that we don't land this change on hackathon-2020-11 (which is master, and I'd prefer not to maintain a separate branch) until we're confident that it doesn't significantly slow down the PR turnaround time.

To improve this situation in general, I'd suggest putting in a PR to split the big CI tasks some more to speed it up again.  I once had it down to 8-9 minutes... maybe we can remove the Node 12.x (latest) tests and just keep 12.14.1?

**Review by michaelfig — APPROVED** [review]
Sorry, didn't mean to hold this up so long.  FWIW, the github workflows change was a purely mechanical review (just had to compare the `BOILERPLATE` with other jobs).

LGTM!

File: .github/workflows/test-all-packages.yml
Comment (michaelfig) [inline .github/workflows/test-all-packages.yml]: @FUDCo I think this is the magic you want not to run on every pull request:

```suggestion
    # Only run if we were a push to master.
    if: ${{ github.event_name == 'push' }}
    needs: build
```

Comment (katelynsills) [reply .github/workflows/test-all-packages.yml]: Hmm, why not split it out into its own file and change the conditions by removing `pull_request` below:

```
on:
 push:
   branches: [master]
 pull_request:
```

Comment (michaelfig) [reply .github/workflows/test-all-packages.yml]: > Hmm, why not split it out into its own file and change the conditions by removing `pull_request` below:

Then it can't depend on the build task, and I'm not sure it can leverage any of the caches from test-all-packages.

**Pull request merged**
